File: BlueTelephone/Background/Packet.cs
namespace BlueTelephone.Background
{
    /// <summary>
    /// 패킷 형태
    /// </summary>
    public class Packet
    {
        /// <summary>
        /// 타임스탬프,
        /// 활용 방안은 미정
        /// </summary>
        public required string TS { get; set; }
        /// <summary>
        /// 메시지 코드,
        /// 백그라운드와 공유
        /// </summary>
        public required int MsgCode { get; set; }
        /// <summary>
        /// 메시지,
        /// 문자열의 리스트로 이루어짐
        /// </summary>
        public required List<string> Msg { get; set; }
    }
}

File: background/packet.go
package main

import (
	"encoding/json"
	"log"
	"net"
	"time"
)

// 패킷 구조체
type Packet struct {
	// 타임스탬프
	TS string
	// 메시지 코드
	MsgCode int
	// 메시지
	Msg []string
}

// 폼으로 패킷을 전달하는 함수
func WritePacket(conn net.Conn, msgCode int, msg []string) {
	packet := &Packet{
		TS:      time.Now().String(),
		MsgCode: msgCode,
		Msg:     msg,
	}

	json, err := json.Marshal(packet)

	if err != nil {
		log.Fatalln(err)
	}

	buf := make([]byte, 1024)
	copy(buf, []byte(string(json)))

	_, err = conn.Write(buf)

	if err != nil {
		log.Fatalln(err)
	}
}
